src/store/configureStore.ts

```diff
@@ -1,39 +1,28 @@
-import type { AppState } from "../types";
-
-import { createStore, applyMiddleware, compose } from "redux";
 import createSagaMiddleware from "redux-saga";
 import createRootReducer from "../reducers";
 import rootSaga from "../sagas";
 import { createHashHistory } from "history";
-
+import { configureStore } from "@reduxjs/toolkit";
 import { routerMiddleware } from "connected-react-router";
 
 const sagaMiddleware = createSagaMiddleware();
 
 export const hashHistory = createHashHistory();
 
-export default function configureStore(
-  initialState: Object = {},
-  history = hashHistory
-) {
-  const finalCreateStore = compose(
-    applyMiddleware<AppState, any, any>(
-      sagaMiddleware,
-      routerMiddleware(history)
-    ),
-    // prettier-ignore
-    (window as any).__REDUX_DEVTOOLS_EXTENSION__
-      ? ((window as any).__REDUX_DEVTOOLS_EXTENSION__() as () => void)
-      : (f) => f
-  )(createStore);
-
-  const store = finalCreateStore(
-    createRootReducer(history),
-    initialState as any
-  );
+export function configureAppStore(initialState = {}, history = hashHistory) {
+  const store = configureStore({
+    reducer: createRootReducer(history),
+    middleware: getDefaultMiddleware =>
+      getDefaultMiddleware({
+        thunk: false,
+      }).concat(sagaMiddleware, routerMiddleware(history)),
```

Comment:
The `defaultMiddleware` includes some nice additions including devtools without manual setup and warnings when save something to the store that isn't serializable.